core/integer/divide_spec.rb

```diff
@@ -1,7 +1,9 @@
 require File.expand_path('../../../spec_helper', __FILE__)
 require File.expand_path('../shared/arithmetic_exception_in_coerce', __FILE__)
 
-describe "Integer#/" do
-  it_behaves_like :integer_arithmetic_exception_in_coerce, :/
+ruby_version_is '2.4' do
+  describe "Integer#/" do
+    it_behaves_like :integer_arithmetic_exception_in_coerce, :/
```

Comment:
Can the guards not be part of the shared example?

Reply:
By having `describe "Integer#/"` I think it's OK to mean it's `Fixnum#/` + `Bignum#/` behavior in Ruby 2.3 and before.
No need to separate those specs, as I meant in #578.

Reply:
@eregon Do you mean we need to "adopt" Bignum specs as well?

Please correct me if I misunderstood you. I would like to copy/paste (and revise) all Fixnum/Bignum specs to Integer specs with ruby version guard "2.4" and add ruby version guard ""..."2.4" to existing Fixnum/Bignum specs.

Reply:
Yes, really move (and merge) `core/fixnum/*` and `core/bignum/*` to `core/integer` and rename descriptions to `Integer#some_method`.
So no extra version guards.
The version guards are not needed and not worth the trouble just for classifying methods differently.

I think duplicating the specs would be bad.
And making them shared specs would make them needlessly complicated.

Please share your thoughts if you disagree.
Maybe it's better to wait 2.2 and 2.3 become unsupported, so then it's really clear there should no longer be `core/{fixnum,bignum}` specs.
I'm proposing to do that already now since anyway behaviorally it is compatible, and the new unified Integer has a nicer structure than separate "small/big integers" specs.

Reply:
@eregon FYI There are following minor tdifferences:
  - `Fixnum#hash` (as well as `Integer#hash`) is implemented in `Kernel`
  - `Fixnum#coerce` is implemented in `Numeric`
  - `Fixnum#eql?` (as well as `Integer#eql?`) is implemented in `Numeric`

I am OK with moving specs and completely finished it

Reply:
- `Fixnum#coerce` is implemented in `Numeric`

It's in `Integer` in 2.5 I think:
```ruby
> 3.method :coerce
=> #<Method: Integer#coerce>
```